### PR TITLE
fix: Extract the models and images content from subfolders to the root level

### LINKED ISSUE #2282: Changing male only to female only TPW adds a non existent content to the representation
## 🎉 Description

Changing the representation of a third party wearable (maybe a standard wearable too, need to check) that is male only to female only ends up with a representation that has a content in the contents array that is `female/male/` and that is not part of the contents, so the schema validation at the catalysts blows up

* [x] 🐛 This is a bug report.
* [ ] 📈 This is a feature request.

## 📝 Details

When deploying the entity to the catalyst, it returns this error:

```
{"errors":["Representation content: 'female/male/' is not one of the content files"]}
```

### PR DESCRIPTION
**Pull request by cyaiox**
This PR cleans the model content when uploading a .zip file with male/female representations and processes the file structure to save the proper representation schema in DB.

Closes #2282

### CODE REVIEW

File: src/components/Modals/CreateSingleItemModal/CreateSingleItemModal.tsx [hunk 1]
```diff
@@ -151,8 +152,170 @@ export default class CreateSingleItemModal extends React.PureComponent<Props, St
     return { male, female, all }
   }
 
+  sortContentZipBothBodyShape = (contents: Record<string, Blob>): SortedContent => {
```
Comment (cyaiox) [inline src/components/Modals/CreateSingleItemModal/CreateSingleItemModal.tsx:155]: This method sorts the content from a .zip file containing both representations.

File: src/components/Modals/CreateSingleItemModal/CreateSingleItemModal.tsx [hunk 2]
```diff
@@ -151,8 +152,170 @@ export default class CreateSingleItemModal extends React.PureComponent<Props, St
...
+        itemSortedContents: sortedContents.all,
+        view: CreateItemView.THUMBNAIL,
+        fromView: CreateItemView.THUMBNAIL
+      })
+    } else {
+      onSave(item as Item, sortedContents.all)
+    }
+  }
+
+  addItemRepresentation = async (sortedContents: SortedContent, representations: WearableRepresentation[]) => {
```
Comment (cyaiox) [inline src/components/Modals/CreateSingleItemModal/CreateSingleItemModal.tsx:242]: Extract the add item representation logic into a method

File: src/components/Modals/CreateSingleItemModal/CreateSingleItemModal.tsx [hunk 3]
```diff
@@ -151,8 +152,170 @@ export default class CreateSingleItemModal extends React.PureComponent<Props, St
...
+      },
+      updatedAt: +new Date()
+    }
+
+    // Do not change the thumbnail when adding a new representation
+    delete sortedContents.all[THUMBNAIL_PATH]
+    onSave(item, sortedContents.all)
+  }
+
+  modifyItem = async (pristineItem: Item, sortedContents: SortedContent, representations: WearableRepresentation[]) => {
```
Comment (cyaiox) [inline src/components/Modals/CreateSingleItemModal/CreateSingleItemModal.tsx:271]: Extract the `modify item` logic into a method

File: src/components/Modals/CreateSingleItemModal/CreateSingleItemModal.tsx [hunk 4]
```diff
@@ -502,6 +529,31 @@ export default class CreateSingleItemModal extends React.PureComponent<Props, St
     return representations
   }
 
+  buildRepresentationsZipBothBodyshape(contents: SortedContent): WearableRepresentation[] {
```
Comment (cyaiox) [inline src/components/Modals/CreateSingleItemModal/CreateSingleItemModal.tsx:532]: This method builds the representations from a .zip file containing both representations.

File: src/modules/item/utils.ts [hunk 5]
```diff
@@ -87,6 +110,10 @@ export function getModelPath(representations: WearableRepresentation[]) {
   return representations[0].mainFile
 }
 
+export function getModelFileNameFromSubfolder(modelPath: string) {
```
Comment (cyaiox) [inline src/modules/item/utils.ts:113]: This method gets the model filename from a subfolder

**Review by cazala — APPROVED** [review]
🙌 ✅

File: src/modules/item/utils.ts [hunk 6]
```diff
@@ -61,6 +61,29 @@ export function getBodyShapeType(item: Item): BodyShapeType {
   }
 }
 
+export function getBodyShapeTypeFromContents(contents: Record<string, Blob>) {
+  let hasMale = false,
+    hasFemale = false
+
+  Object.keys(contents).forEach((key: string) => {
```
[the remark this answers is not in the document]
Comment (cyaiox) [reply src/modules/item/utils.ts:68]: sure!

**Pull request merged**